lib/connextclient/ethprovider.ts

```diff
@@ -0,0 +1,60 @@
+import { Observable, from, combineLatest } from 'rxjs';
+import { ethers, BigNumber } from 'ethers';
+import { mergeMap } from 'rxjs/operators';
+
+type OnChainTransaction = {
+  nonce: number;
+  gasPrice: BigNumber;
+  gasLimit: BigNumber;
+  to: string;
+  value: BigNumber;
+  data: string;
+  chainId: number;
+  v: number;
+  r: string;
+  s: string;
+  from: string;
+  hash: string;
+};
+
+const onChainSendERC20 = (
+  host: string,
+  port: number,
+  chainId: number,
+  seed: string,
+  contractAddress: string,
+  destinationAddress: string,
+  units: string,
+): Observable<OnChainTransaction> => {
+  const provider = new ethers.providers.JsonRpcProvider(
+    {
+      url: `http://${host}:${port}/ethprovider/${chainId}`,
+    },
+    {
+      name: 'simnet',
+      chainId,
+    },
+  );
+  const erc20abi = ['function balanceOf(address) view returns (uint)', 'function transfer(address to, uint amount)'];
+  const signer = ethers.Wallet.fromMnemonic(seed).connect(provider);
+  const erc20 = new ethers.Contract(contractAddress, erc20abi, signer);
```

Comment:
Call my old fashioned but I like classes with constructors that initialize objects like the `provider` and simple function that only have a couple arguments. But you do you :shrug:

Reply:
I'm planning to refactor this function so that it receives signer (with provider) as an argument.